src/weights.jl

```diff
@@ -549,15 +545,15 @@ Compute the weighted quantiles of a vector `x` at a specified set of probability
 values `p`, using weights given by a weight vector `w` (of type `AbstractWeights`).
 Weights must not be negative. The weights and data vectors must have the same length.
 
-The quantile for `p` is defined as follows. Denoting
-``S_k = (k-1)w_k + (n-1) \\sum_{i<k}w_i``, define ``x_{k+1}`` the smallest element of `x`
-such that ``S_{k+1}/S_{n}`` is strictly superior to `p`. The function returns
-``(1-\\gamma) x_k + \\gamma x_{k+1}`` with  ``\\gamma = (pS_n- S_k)/(S_{k+1}-S_k)``.
-
-This corresponds to  R-7, Excel, SciPy-(1,1) and Maple-6 when `w` contains only ones
-(see [Wikipedia](https://en.wikipedia.org/wiki/Quantile)).
+With frequency weights, the function returns the same result as `quantile` for a vector with repeated values.
+With non frequency weights,  denote N the length of the vector, w the vector of weights normalized to sum to 1, `h = p (N - 1) + 1`  and ``S_k = 1 + (k-1) * wk + (N-1) \\sum_{i<=k}w_i/\\sum_{i<=N}w_i``, define ``x_{k+1}`` the smallest element of `x` such that ``S_{k+1}`` is strictly superior to `h`. The function returns
+``x_k + \\gamma (x_{k+1} -x_k)`` with  ``\\gamma = (h - S_k)/(S_{k+1}-S_k)``
+In particular, when `w` is a vector of one, the function returns the same result as `quantile`.
 """
-function quantile(v::RealVector{V}, w::AbstractWeights{W}, p::RealVector) where {V,W<:Real}
+
```

Comment:
No line breaks before nor after the signature. I think the style of this package is not to have spaces after commas for type parameters.